dtfs-central-api/src/services/utilisation-report-data-validator/index.ts

```diff
@@ -99,38 +99,35 @@ export const validateUtilisationReportCsvCellData = (
     generatePaymentExchangeRateError,
   ];
 
-  return csvData.flatMap((csvRow) => {
-    const csvDataErrors: UtilisationReportDataValidationError[] = [];
-
-    cellValidations.forEach(({ header, errorGenerator }) => {
-      if (availableHeaders.includes(header)) {
-        const error = errorGenerator(csvRow[header], csvRow.exporter?.value);
-        if (error) {
-          csvDataErrors.push(error);
+  const errors = await Promise.all(
+    csvData.map(async (csvRow) => {
+      const cellErrorPromises = cellValidations.map(({ header, errorGenerator }) => {
+        if (!availableHeaders.includes(header)) {
+          return null;
         }
-      }
-    });
 
-    rowValidations.forEach((errorGenerator) => {
-      const error = errorGenerator(csvRow);
-      if (error) {
-        csvDataErrors.push(error);
-      }
-    });
+        return errorGenerator(csvRow[header], csvRow.exporter?.value);
+      });
 
-    return csvDataErrors;
-  });
+      const rowErrorPromises = rowValidations.map((errorGenerator) => errorGenerator(csvRow));
+
+      return Promise.all([...cellErrorPromises, ...rowErrorPromises]);
+    }),
+  );
+
+  // Filter out null errors so that we don't return any null elements in the response array
+  return errors.flat().filter((error): error is UtilisationReportDataValidationError => error !== null);
```

Comment:
Old comment from @Zainzzkk when you can please @RLavender98 , quick one

"can this be a const and return the const"

Reply:
Ahh ok I undid this by accident resolving other comments since I thought it was nicer and hadn't seen it was done in response to this comment.